ddl/index.go

```diff
@@ -531,6 +543,15 @@ func (w *addIndexWorker) getIndexRecord(handle int64, recordKey []byte, rawRecor
 		return nil, errors.Trace(err)
 	}
 	idxVal := make([]types.Datum, len(idxInfo.Columns))
+	if len(w.colExprMap) != 0 {
+		for _, col := range w.table.Cols() {
+			ri, ok := w.rowMap[col.ID]
+			if ok {
+				w.rowSlice[col.Offset] = ri
+			}
+		}
+		w.row = chunk.MutRowFromDatums(w.rowSlice).ToRow()
```

Comment:
Since `rowSlice` is only used here, and the check are both `if len(colExprMap) != 0`. So let it be a local variable instead of one in struct.